src/components/ha-qr-scanner.ts

```diff
@@ -278,16 +312,40 @@ class HaQrScanner extends LitElement {
         message,
       },
     });
-    this.error = undefined;
+    this._warning = undefined;
+    this._error = undefined;
   }
 
   private _reportError(message: string) {
-    fireEvent(this, "qr-code-error", { message });
+    const canvas = this._qrScanner?.$canvas;
+    if (canvas) {
+      canvas.style.display = "none";
+    }
+    this._error = message;
+  }
+
+  private _reportWarning(message: string) {
+    this._warning = message;
+    this._notifyExternalScanner(message);
```

Comment:
Note (not related to this change), in `_notifyExternalScanner` we should actually check for `_nativeBarcodeScanner` instead of `this.hass.auth.external`.

Nice would be I think to check in this function if we are native:
```ts
if (this._nativeBarcodeScanner) {
	this._notifyExternalScanner(message);
} else {
	this._warning = message;
}
```